### DISCUSSION TITLE
Git & VS Code connection

### DISCUSSION THREAD

**Discussion by mohanbeckford** [community/community#56146] [Actions] [answered]
### Select Topic Area

Question

### Body

I'm trying to clone a repository from GitHub into VS Code and i am getting the following error

/Users/user/.ssh/config line 1: no argument after keyword "host*"
/Users/user/.ssh/config: terminating, 1 bad configuration options
fatal: Could not read from remote repository.

**Comment by airtower-luna** [comment]
What that tells me is that a) you're trying to clone over SSH and b) your SSH config is invalid. `host*` sounds like a typo in a rule that's supposed to apply to all hosts (missing a space before the `*`). Though such a rule rarely makes sense, usually you'd better set global options instead. But saying exactly how to fix it would require seeing your SSH config. :thinking:

**Reply by mohanbeckford** [reply]
This i believe is the config file

HOST*
IgnoreUnknown UseKeychain
ADDKeysToAgent yes
UseKeychain yes
IdentityFile ~/.SSH/id_ed25519

**Reply by airtower-luna** [reply]
It does have that same error, yes. If you want those options to apply to SSH connections to *any* host, just remove the `HOST*` line.

If you want to have the rules apply only for specific hosts, make sure to leave a space between `Host` and the following names/patterns.

**Reply by mohanbeckford** [reply]
I put a space as you suggested between host and * and I got this

root@Nexts-MacBook-Pro bootcamp # git clone git@github.com:mohanbeckford/prework-study-guide.git
Cloning into 'prework-study-guide'...
The authenticity of host 'github.com (140.82.114.4)' can't be established.
ED25519 key fingerprint is SHA256:+DiY3wvvV6JhbpZisF/zLDA0zPMSvHdkr4UvCOqU.
This key is not known by any other names
Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
Warning: Permanently added 'github.com' (ED25519) to the list of known hosts.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

**Reply by airtower-luna** [reply]
1. You should *not* be running Git as `root`. Git doesn't do anything that'd require root access, doing it is an unnecessary security risk. :warning: 
2. I hope you [checked the fingerprint](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/githubs-ssh-key-fingerprints) before accepting the server key. :slightly_smiling_face: 
3. "Permission denied" means you didn't have a suitable key to log in to GitHub. This could be a consequence of point 1 (because *your* key is not the key of `root`). If you have trouble solving that, run `ssh -vT git@github.com` to get debug info on the SSH connection itself.

**Reply by mohanbeckford** [reply]
Is there a way to wipe all of this out and re do it?

**Reply by airtower-luna** [reply, marked as answer]
Which part? :wink: 

1. For not running as root, simply run Git as your usual user.
2. For checking the fingerprint, you can delete the known one with `ssh-keygen -R github.com`. Then connect again and check when asked.
3. There's nothing to *re*-do here. You might need to add your public key to your account. If you want to replace your existing key you can just generate a new one.

**Comment by mohanbeckford** [comment]
I figured it out. I had to add the GitHub public fingerprint
